skimage/util/dtype.py

```diff
@@ -596,5 +640,183 @@ def img_as_bool(image, force_copy=False):
     The upper half of the input dtype's positive range is True, and the lower
     half is False. All negative values (if present) are False.
 
+    """
+    return rescale_to_bool(image, force_copy=force_copy)
+
+
+def rescale_to_float32(image, *, force_copy=False):
+    """Convert an image to single-precision (32-bit) floating point format.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    force_copy : bool, optional
+        Force a copy of the data, irrespective of its current dtype.
+
+    Returns
+    -------
+    out : ndarray of float32
+        Output image.
+
+    Notes
+    -----
+    The range of a floating point image is [0.0, 1.0] or [-1.0, 1.0] when
+    converting from unsigned or signed datatypes, respectively.
```

Comment:
Maybe we can use this opportunity to get rid of the confusing notion of an image scaled [-1, 1], and just stick with [0, 1]?

Reply:
That would be worthwhile. The problem is that the `rescale_to_float*` functions are used internally in other places. We'd still have to make sure that the new behavior wouldn't affect results and at that point.

However, it might be more confusing to introduce the new names now with the old behavior, and then do a hard switch for skimage2. 

Maybe we use a private function internally to ensure other functions aren't affected by this? Thoughts?

Reply:
How about if we introduce an argument `legacy_signed_range` (or a better name) to recover the old [-1, 1] behavior for signed ints, but it defaults to `False` ([0, 1]). Users can easily and more importantly quickly (in our case) migrate and the name communicates that this behavior is legacy. Then in skimage2, we deprecate `legacy_signed_range`.

Reply:
But I would suggest doing that in a follow-up PR.

Reply:
Sounds, good.

Reply:
OK with legacy_range, and switching for 2.0.

We should start our migration guide so we can document these types of things users may need to know.

Reply:
Addressed in https://github.com/scikit-image/scikit-image/pull/7697/commits/5d231c40e2bec44ebd11b56b946453de1e3ed847 and following.